### PR TITLE
initial tool api

### PR DESCRIPTION
**Pull request by unix-supremacist**
initial tool api, probably needs some slight changes but is ready for review

### CODE REVIEW

File: felt-tool-api/src/main/java/io/github/feltmc/feltapi/api/tool/FeltPickaxeItem.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    public FeltPickaxeItem(ToolMaterial material, int attackDamage, float attackSpeed, Settings settings, boolean torching) {
+        super(material, attackDamage, attackSpeed, settings);
+        torchingEnabled = torching;
+    }
+
+    @Override
+    public ActionResult useOnBlock(ItemUsageContext context) {
+        if (torchingEnabled){
```
Comment (PseudoDistant) [inline felt-tool-api/src/main/java/io/github/feltmc/feltapi/api/tool/FeltPickaxeItem.java:22]: Yes

**Comment by PseudoDistant** [comment]
Have you tested this?

**Review by PseudoDistant — APPROVED** [review]
:+1:

**Comment by unix-supremacist** [comment]
it's all been tested in isolation

**Comment by unix-supremacist** [comment]
oh nice

**Pull request merged**